FreeRTOS/FreeRTOS-Kernel
# Coverity Scan Workflow Fix

Description
-----------
Currently the Coverity Scan attempts to run on every fork that pulls the file.
* This leads to anybody who pulls this file getting emails that their workflow failed to run when the cron job attempts to run.
Set the scan to only run if the repo is `FreeRTOS/FreeRTOS-Kernel`
Change the scan from a cron job to a job that runs on commit to mainline, or if triggered manually.

Test Steps
-----------
Tests for this change can be seen at this PR I raised to my own repo.
https://github.com/Skptak/FreeRTOS-Kernel/pull/3

Proof that this will cause a coverity scan to be triggered when a PR is merged can be seen [here](https://github.com/Skptak/FreeRTOS-Kernel/actions/runs/6869487837)
    * **NOTE:** The reason this run is failing is because I don't have the necessary token and other secret on my own fork


Proof that the action will not run unless from FreeRTOS/FreeRTOS-Kernel can be seen [here](https://github.com/Skptak/FreeRTOS-Kernel/actions/runs/6869609168)

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
